prov/verbs/src/fi_verbs.h

```diff
@@ -100,6 +102,8 @@
 #define VERBS_INJECT_FLAGS(ep, len, flags) ((((flags) & FI_INJECT) || \
 		len <= (ep)->inject_limit) ? IBV_SEND_INLINE : 0)
 #define VERBS_INJECT(ep, len) VERBS_INJECT_FLAGS(ep, len, (ep)->info->tx_attr->op_flags)
+#define VERBS_INJECT_DESC(ep, len, desc) \
+	(MR_DESC_IFACE(desc) == FI_HMEM_SYSTEM ? VERBS_INJECT(ep, len) : 0)
```

Comment:
It's preferable to pass desc to the INJECT and INJECT_FLAGS macro, and update INJECT_FLAGS with the iface check.  This ensures that iface is handled properly at the lowest level.  As written, the developer has to somehow know not to use either of the first two macros, since it does not correctly handle the iface.

Reply:
Will fix.